lib/livebook/session/data.ex

```diff
@@ -1727,6 +1741,17 @@ defmodule Livebook.Session.Data do
     set!(data_actions, notebook: %{data.notebook | hub_secret_names: hub_secret_names})
   end
 
+  defp update_notebook_deployment_group({data, _} = data_actions, hub) do
```

Comment:
When deployment group changes we need to sync secrets accordingly. And we need to consider deployment group for the initial secrets. @josevalim correct me if I'm wrong!

Reply:
Ah, does it only have effect for deployment? Should we have a way to set it for the session while developing?

Reply:
Yes, it only impacts deployment and nothing more. We should soon redesign the deployment interface.